### ISSUE TITLE
using ModalDamping will slow down the dynamic calculation process 

### ISSUE THREAD

**Issue by HaoxinGuo** [OpenSees/OpenSees#253]
When I use Newmark integrator to solve structure with modalDamping, It will very slowly. When I check it ,I found the reason is the function `addModalDampingMatrix()` . this function will test the CFactor ==0 or not 
```
double cFactor=this->getCFactor();
  if (cFactor == 0)
    return 0;
```
when using newmark integrator, the CFactor not equal to 0. So it will run the rest of the code，and it costs a lot of time.
But I forcethe cFactor==0 ,the solution is also right. 
So what does this function do？

**Comment by mhscott** [comment]
The value returned by Newmark::getCFactor is c2, which is the constant applied to the velocity in the time integration, regardless of the type of damping used.  I don't see how setting it to zero will get you to the same solution, unless perhaps it's getting to the same solution but taking more Newton iterations to get there.

**Comment by HaoxinGuo** [comment]
> The value returned by Newmark::getCFactor is c2, which is the constant applied to the velocity in the time integration, regardless of the type of damping used. I don't see how setting it to zero will get you to the same solution, unless perhaps it's getting to the same solution but taking more Newton iterations to get there.

You are right. However the CentralDifference don't have this function in CentralDifference.cpp. Andthe method also has the constant $1/2 \Delta t$ ，as well as the BackwardEuler interator.
Why?

**Comment by zhuminjie** [comment]
These constants are different for different time integrators. It usually leaves to the integrator to decide how to define their constants applied to displacements, velocities, and accelerations. That's why we need all different kinds of integrator classes, which implement the time stepping algorithms.

**Comment by HaoxinGuo** [comment]
So aren't all integrators suitable for modaldamping now?
We should  add function ,such as `getCFactor()`, to the inherited  integrator  class by ourselves，if needed.
When I remove the code `getCFactor()` in  inherited  integrator ,it will call parent class,and the parent 
  function `getCFactor()`  always returns 0.
So I think the function `addModalDampingMatrix()`  is useless. You can test it in integrator newmark. like this:
remove the codes in newmark.cpp
```
//double
//Newmark::getCFactor(void) {
//return c2;
}
```
as well as remove code in newmark.h
```
  double getCFactor(void);
```
compile the project ,then run the source code. You will find the answer is same with not romoving the code above get .

**Comment by jaabell** [comment]
Any updates on this? I had noticed that models with damping incurred in a weird performance hit.

**Comment by fmckenna** [comment]
try modalDampingQ . the Q stands for Quick .. does not bother figuring C matrix contributions (no contributions to LHS of equation) ; it just makes sure right hand side is correct.
